### PR TITLE
Initial version

### LINKED ISSUE #27: Executable uber-jar container image how-to
Package the app as executable uber-jar
Create Dockerfile
Run docker CLI

### PR DESCRIPTION
**Pull request by tsegismont**
Closes vertx-howtos/vertx-howtos.github.io#27

Initial version of the executable-jar-docker-howto

### CODE REVIEW

File: docker-gradle/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# 1st Docker build stage: build the project with Gradle
+FROM gradle:6.8.3-jdk11 as builder
+WORKDIR /project
+COPY . /project/
+RUN gradle assemble --no-daemon
+
+# 2nd Docker build stage: copy builder output and configure entry point
+FROM openjdk:11-slim
```
Comment (jponge) [inline docker-gradle/Dockerfile]: How about using https://hub.docker.com/_/adoptopenjdk ?

File: docker-gradle/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ENV APP_DIR /application
+ENV APP_FILE container-uber-jar.jar
+
+EXPOSE 8888
+
+WORKDIR $APP_DIR
+COPY --from=builder /project/build/libs/*-fat.jar $APP_DIR/$APP_FILE
+
+ENTRYPOINT ["sh", "-c"]
+CMD ["exec java -jar $APP_FILE"]
```
Comment (jponge) [inline docker-gradle/Dockerfile:18]:
```suggestion
CMD ["exec", "java", "-jar", "$APP_FILE"]
```

File: docker-maven/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ENV APP_DIR /application
+ENV APP_FILE container-uber-jar.jar
+
+EXPOSE 8888
+
+WORKDIR $APP_DIR
+COPY --from=builder /project/target/*-fat.jar $APP_DIR/$APP_FILE
+
+ENTRYPOINT ["sh", "-c"]
+CMD ["exec java -jar $APP_FILE"]
```
Comment (jponge) [inline docker-maven/Dockerfile:18]: Same remark here

**Review by jponge — CHANGES_REQUESTED** [review]
Minor remarks, otherwise a solid how-to!

**Comment by tsegismont** [comment]
@jponge PTAL

We must use the string for `CMD` because it the argument for the `ENTRY_POINT` (`bash -c`)

**Comment by jponge** [comment]
Good catch indeed

**Review by jponge — APPROVED** [review]

**Pull request merged**